### PR TITLE
Update snapping GUI units when canvas CRS changes

### LINKED ISSUE #31961: Snapping map units always show "meters even when the project is in different units
**Describe the bug**

On the snapping toolbar, you have to ways of setting the snapping tolerance "pixels" and map units. The "map units" option should show the actual units, but it seems to always show the units as "meters" even if the project units are, for example "degrees". Moreover, the actual snapping accurs using the project units and not meters.

**How to Reproduce**

1. Open a new project and set the projection to 4326
2. Add a new temporary layer in 4326
3. Enable snapping and you will see that besides pixels the other option is meters.
4. keep the tolerance as 12
4. Try digiting something and you will see the snapping accurs at 12 degrees, and not 12 meters.

image

**QGIS and OS versions**

QGIS version | 3.9.0-Master | QGIS code revision | cebdc8dded
-- | -- | -- | --
Compiled against Qt | 5.9.5 | Running against Qt | 5.9.5
Compiled against GDAL/OGR | 2.2.3 | Running against GDAL/OGR | 2.2.3
Compiled against GEOS | 3.6.2-CAPI-1.10.2 | Running against GEOS | 3.6.2-CAPI-1.10.2 4d2925d6
Compiled against SQLite | 3.22.0 | Running against SQLite | 3.22.0
PostgreSQL Client Version | 11.5 (Ubuntu 11.5-1.pgdg18.04+1) | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.2
PROJ.4 Version | 493
OS Version | Ubuntu 18.04.3 LTS

**Additional context**

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #31961

### CODE REVIEW

File: src/app/qgsidentifyresultsdialog.cpp [hunk 1]
```diff
@@ -618,14 +618,17 @@ void QgsIdentifyResultsDialog::addFeature( QgsVectorLayer *vlayer, const QgsFeat
...
+      valueLabel->setWordWrap( true );
       valueLabel->setOpenExternalLinks( true );
       attrItem->setData( 1, Qt::DisplayRole, QString() );
       attrItem->treeWidget()->setItemWidget( attrItem, 1, valueLabel );
     }
     else
     {
-      attrItem->setData( 1, Qt::DisplayRole, value );
-      attrItem->treeWidget()->setItemWidget( attrItem, 1, nullptr );
+      auto valueLabel { qgis::make_unique<QLabel>( value ) };
```
Comment (nyalldawson) [inline src/app/qgsidentifyresultsdialog.cpp]: This looks unrelated?

Comment (elpaso) [reply src/app/qgsidentifyresultsdialog.cpp]: What do you mean? That a unique pointer is not required here and a plain `new` would do just fine?

Comment (elpaso) [reply src/app/qgsidentifyresultsdialog.cpp]: @nyalldawson gentle ping

Comment (nyalldawson) [reply src/app/qgsidentifyresultsdialog.cpp]: I mean I can't see what it's got to do with snapping settings...?

Comment (elpaso) [reply src/app/qgsidentifyresultsdialog.cpp]: Oh sorry, I see it's a leftover from this https://github.com/qgis/QGIS/issues/31650#issuecomment-535090459  , I'll remove it.

**Comment by nyalldawson** [comment]
LGTM! Can you remove the backport label?
